.github/workflows/ci.yml

```diff
@@ -17,36 +17,40 @@ jobs:
       matrix:
         platform: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.platform }}
+    permissions:
+      contents: read
 
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
```

Comment:
I don't think there's a need for this change, in particular because of "security reasons".

> Downside is that the updates must be done by updating the commit hash instead of it being done automatically through moving the tag to a new release.
This can be solved by using tools like Dependabot or Renovatebot by adding a comment of the version used, enabling automated dependency updates. Tools like StepSecurity can also be used.

Are we "solving it" per your suggestion in https://github.com/MarkBind/markbind/issues/2488#issuecomment-2041400560? If so it should be part of this PR.

Reply:
Hi @tlylt, thanks for the review, I have added the dependabot file that will check versions weekly.

Reply:
Thanks @KevinEyo1.

I still have concerns with the effort to benefit ratio of these change. I don't think dependabot PRs will get enough attention especially after the semester, so it's a maintenance burden that we are creating...

I will leave it to MarkBind active devs to approve/comment.

Reply:
Hmmm I think I wouldn't mind using the hash instead of the version number. 
If we need to update, we have to change the number anyway. The only extra steps involved is we have to go to the repository (which we should be doing anyway when doing an update) and obtain the new SHA. 
I think the benefit of better security is higher than the cost, especially with more dev ops that we are adding in thanks to @KevinEyo1.

Reply:
> If we need to update, we have to change the number anyway.

That's only for major version bump. For non breaking changes, using v.x syntax automatically uses the latest release of the action. So no "change the number" required.

Reply:
Ah i see! Thanks i did not know that. 
In that case, I think keeping it as it is makes more sense. I feel like there will be more issues (including security issues) with not keeping updated with the minor version bumps..